runner/operator/roslin_operator/bin/test_pair_request.py

```diff
@@ -0,0 +1,211 @@
+        "tumor_type": "Normal"
+        },
+        {
+        "patient_id": "C-W86LMR",
+        "bait_set": "IMPACT468_BAITS",
+        "tumor_type": "Tumor"
+        }
+        ]
+        pairs = compile_pairs(samples)
+        expected_pairs = {
+        'tumor': [
+        {'patient_id': 'C-W86LMR', 'bait_set': 'IMPACT468_BAITS', 'tumor_type': 'Tumor'}
+        ],
+        'normal': [
+        {'patient_id': 'C-W86LMR', 'bait_set': 'IMPACT468_BAITS', 'tumor_type': 'Normal'}
+        ]
+        }
+        self.assertTrue(pairs == expected_pairs)
+
+    def test_compile_pairs2(self):
+        """
+        Test pairing with multiple samples in a request
+        """
+        samples = [
+        {
+        "bait_set": "IMPACT468_BAITS",
+        "patient_id": "C-DRKHP7",
+        "tumor_type": "Normal"
+        },
+        {
+        "bait_set": "IMPACT468_BAITS",
+        "patient_id": "C-8VK0V7",
+        "tumor_type": "Normal"
+        },
+        {
+        "bait_set": "IMPACT468_BAITS",
+        "patient_id": "C-DRKHP7",
+        "tumor_type": "Tumor"
+        },
+        {
+        "bait_set": "IMPACT468_BAITS",
+        "patient_id": "C-8VK0V7",
+        "tumor_type": "Tumor"
+        },
+        {
+        "bait_set": "IMPACT468_BAITS",
+        "patient_id": "C-DRKHP7",
+        "tumor_type": "Tumor"
+        }
+        ]
+        pairs = compile_pairs(samples)
+        expected_pairs = {
+        'tumor': [
+        {'bait_set': 'IMPACT468_BAITS', 'patient_id': 'C-DRKHP7', 'tumor_type': 'Tumor'}, {'bait_set': 'IMPACT468_BAITS', 'patient_id': 'C-8VK0V7', 'tumor_type': 'Tumor'}, {'bait_set': 'IMPACT468_BAITS', 'patient_id': 'C-DRKHP7', 'tumor_type': 'Tumor'}
+        ],
+        'normal': [
+        {'bait_set': 'IMPACT468_BAITS', 'patient_id': 'C-DRKHP7', 'tumor_type': 'Normal'}, {'bait_set': 'IMPACT468_BAITS', 'patient_id': 'C-8VK0V7', 'tumor_type': 'Normal'}, {'bait_set': 'IMPACT468_BAITS', 'patient_id': 'C-DRKHP7', 'tumor_type': 'Normal'}
+        ]
+        }
+        self.assertTrue(pairs == expected_pairs)
+
+    def test_compile_pairs3(self):
+        """
+        Test pairing with only a single Normal sample
+        """
+        samples = [
+        {
+        "bait_set": "IMPACT468_BAITS",
+        "patient_id": "C-DRKHP7",
+        "tumor_type": "Normal"
+        }
+        ]
+        pairs = compile_pairs(samples)
+        expected_pairs = {'tumor': [], 'normal': []}
+        self.assertTrue(pairs == expected_pairs)
+
+    def test_compile_pairs4(self):
+        """
+        Test pairing with only a single unpaired Tumor sample
+        Test that the appropriate Normal sample is found from the other samples in the same request
+        missing normal for sample 10075_D_1; querying patient C-DRKHP7
+        """
+        # Load fixtures
+        test_files_fixture = os.path.join(settings.TEST_FIXTURE_DIR, "10075_D.file.json")
+        call_command('loaddata', test_files_fixture, verbosity=0)
```

Comment:
This can't be loaded together with other fixtures same as you did [here](https://github.com/mskcc/beagle/blob/script_dump_request/runner/operator/roslin_operator/bin/test_pair_request.py#L19)? Or there is some other reason you implement it this way?

Reply:
the import [here](https://github.com/mskcc/beagle/blob/script_dump_request/runner/operator/roslin_operator/bin/test_pair_request.py#L19) sets up the base fixtures for all test cases, in order to make the pipeline and file system entries usable. 

This test case is using a specific set of fixtures loaded after that first set to put specific samples' file and filemetadata info into the database.